### PR TITLE
Removes duplicate permissions from webext permission list.

### PR DESCRIPTION
**Pull request by apoorvaeternity**
Fixes https://github.com/mozilla/addons/issues/405

### CODE REVIEW

File: src/olympia/files/models.py [hunk 1]
```diff
@@ -431,8 +431,15 @@ def webext_permissions_list(self):
             return []
         try:
             # Filter out any errant non-strings included in the manifest JSON.
-            return [p for p in self._webext_permissions.permissions
-                    if isinstance(p, basestring)]
+            # Remove any duplicate permissions.
+            permissions = [p for p in self._webext_permissions.permissions
+                           if isinstance(p, basestring)]
+            filtered_permissions = set()
+            filtered_permissions_add = filtered_permissions.add
```
Comment (eviljeff) [inline src/olympia/files/models.py]: Can't you just have `filtered_permissions.add(p)` directly in the list generator?

File: src/olympia/files/models.py [hunk 2]
```diff
@@ -431,8 +431,15 @@ def webext_permissions_list(self):
...
+            return [p for p in permissions if
+                    not (p in filtered_permissions or
+                         filtered_permissions_add(p))]
```
Comment (eviljeff) [inline src/olympia/files/models.py]: I've not tried it, but how does it look if you combine the two generators into one?  So we're not looping through twice.

File: src/olympia/files/models.py (hunk 1 above)
Comment (apoorvaeternity) [inline src/olympia/files/models.py]: I think resolving filtered_permissions.add in each iteration is more costly than resolving a local variable.Am I right?

File: src/olympia/files/models.py (hunk 2 above)
Comment (apoorvaeternity) [inline src/olympia/files/models.py]: Okay I will try to combine them into one.

File: src/olympia/files/models.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/files/models.py]: I don't have benchmarks so you may be right.  But I don't think we'd see the difference in the small numbers of permissions a typical add-on has so I prefer less code instead.

**Comment by apoorvaeternity** [comment]
@eviljeff Thanks for helping me.

**Pull request merged**